modules/json/src/lib/json-configuration.js

```diff
@@ -0,0 +1,35 @@
+// TODO - default parsing code should not be part of the configuration.
+import parseExpressionString from './helpers/parse-expression-string';
+import assert from '../utils/assert';
+const DEFAULT_TYPE_KEY = 'type';
+
+const DEFAULT_CONFIGURATION = {
+  typeKey: DEFAULT_TYPE_KEY,
+  classes: {},
+  reactComponents: {},
+  enumerations: {},
+  // TODO - this needs to be simpler, function conversion should be built in
+  convertFunction
+};
+
+const isObject = value => value && typeof value === 'object';
+
+export default class JSONConfiguration {
+  constructor(configuration) {
+    Object.assign(this, DEFAULT_CONFIGURATION, configuration);
```

Comment:
Should be easy to support backward compatibility by merging `layers` `views` etc into `classes`.